### PR TITLE
Fix race condition when uploading a small package

### PR DESCRIPTION
**Pull request by trygve-lie**
This solves that uploaded files in a package is missing from the upload log and final response of the pkg `PUT` endpoint.

The cause for this is pretty nifty. The reality is that the busboy `finish` event will emit when the writable stream it is has finished consuming all the data which is piped into it. Streams does have an internal back pressure mechanism and when we pipe a package into busboy and from there pipe it into a sink (file writer or google cloud bucket) the back pressure will pause the stream a little bit here and there while the sink create a destination to write too. Due to this writing files to a sink will mostly cause the `finish` event on busboy to emit after all writing is done due to back pressure preventing busboy from being done with emptying the stream.

Though, this plays out different when the file is so small that it fits in the cache of the back pressure. So what happens then is that the package streamed into busboy goes directly into the back pressure cache and it finishes before a sink can even start to write it because the whole uploaded file fits in the back pressure cache.

This solves this by not relying on the busboy `finish` event and instead relying on when we are done unpacking the package.

### CODE REVIEW

File: test/integration/pkg-put-write-Integrity.js [hunk 1]
```diff
@@ -1,116 +0,0 @@
-'use strict';
```
Comment (digitalsadhu) [inline test/integration/pkg-put-write-Integrity.js]: Why did you rename this file to have a capital I?

File: lib/handlers/pkg.put.js [hunk 2]
```diff
@@ -78,15 +79,17 @@ const parser = (sink, incoming) => {
...
+                busboy.emit('completed');
             });
         });
 
         busboy.on('error', error => {
             reject(error);
         });
 
-        busboy.on('finish', () => {
+        busboy.once('completed', () => {
```
Comment (digitalsadhu) [inline lib/handlers/pkg.put.js]: Nice fix.

File: test/integration/pkg-put-write-Integrity.js (hunk 1 above)
Comment (trygve-lie) [reply test/integration/pkg-put-write-Integrity.js]: Its renamed to not have a capital I.
